libr/bin/p/bin_xtr_fatmach0.c

```diff
@@ -92,7 +92,14 @@ static RBinXtrData * extract(RBin* bin, int idx) {
 	arch = r_bin_fatmach0_extract (fb, idx, &narch);
 	if (!arch) return res;
 
-	res = r_bin_xtrdata_new (NULL, NULL, arch->b, arch->offset, arch->size, narch);
+	RBinXtrMetadata *metadata = R_NEW0(RBinXtrMetadata);
+
+	metadata->bits = MACH0_(get_bits)(r_buf_buffer(arch->b));
```

Comment:
This will not work on big endian machines

> On 01 Jul 2016, at 07:36, Aneesh Dogra <notifications@github.com> wrote:
> 
> In libr/bin/p/bin_xtr_fatmach0.c:
> 
> > @@ -92,7 +92,14 @@ static RBinXtrData * extract(RBin* bin, int idx) {
> >  	arch = r_bin_fatmach0_extract (fb, idx, &narch);
> >  	if (!arch) return res;
> >  
> > -	res = r_bin_xtrdata_new (NULL, NULL, arch->b, arch->offset, arch->size, narch);
> > +	RBinXtrMetadata *metadata = R_NEW0(RBinXtrMetadata);
> > +
> > +	metadata->bits = MACH0_(get_bits)(r_buf_buffer(arch->b));
> It takes in a MACH0_(obj_t), which is a structure for macho header fields. It works because we can cast the buffer to that struct. I checked it again here. Works for me!
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>